### PR TITLE
chore(release): upgrade mlmd to 0.25.1

### PR DESCRIPTION
**Pull request by rui5i**
This PR added script on replacing old proto files with new mlmd version and update the generated proto to 0.25.1.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ The script run by `npm run build:protos` can be found at `scripts/gen_grpc_web_p
 The current TypeScript proto library was generated with `protoc-gen-grpc-web` version 1.0.7 with
 `protoc` version 3.10.1.
 
-The Protocol Buffers in `proto/ml_metadata` are taken from version 0.22.1 of the `ml_metadata` proto
+The Protocol Buffers in `proto/ml_metadata` are taken from the master branch(v0.25.1) of the `ml_metadata` proto
```
Comment (Bobgy) [inline README.md]: Did you take it from master branch or v0.25.1?
We should always use a released tag to make sure it's well-tested.

**Review by Bobgy — COMMENTED** [review]
Thanks!
Can you also update https://github.com/kubeflow/frontend/tree/master/proto/ml_metadata or delete that folder + check in the script you use to download it and compile via protoc?

File: README.md (hunk 1 above)
Comment (rui5i) [reply README.md]: Added script on replacing protos. PTAL!

File: src/mlmd/generated/ml_metadata/proto/metadata_store_pb.js [hunk 2]
```diff
@@ -7,9865 +7,10 @@
...
  */
 // GENERATED CODE -- DO NOT EDIT!
+/* eslint-disable */
+// @ts-nocheck
 
 var jspb = require('google-protobuf');
 var goog = jspb;
 var global = Function('return this')();
 
-goog.exportSymbol('proto.ml_metadata.AnyArtifactStructType', null, global);
```
Comment (Bobgy) [inline src/mlmd/generated/ml_metadata/proto/metadata_store_pb.js]: It seems a little weird, this file is now almost empty.

Comment (rui5i) [reply src/mlmd/generated/ml_metadata/proto/metadata_store_pb.js]: Thanks for pointing it out. There was an async bug on previous script which will lead to generate protobuf before file got downloaded. I split the scripts into two parts to have different functions: 
1. replace the proto files with target version(newly added one).
2. generate protobufs(same as before).

Now the protobuf regen steps is updated on README.md with two steps. PTAL!

File: package.json [hunk 3]
```diff
@@ -14,12 +14,14 @@
...
     "lodash.isfunction": "^3.0.9",
-    "react-svg-line-chart": "^2.0.2"
+    "react-svg-line-chart": "^2.0.2",
+    "request": "^2.88.2"
   },
   "scripts": {
     "build": "npx tsc --project tsconfig.lib.json",
     "build:all": "npm run build:protos && npx tsc --project tsconfig.lib.json",
     "build:protos": "node scripts/gen_grpc_web_protos.js",
+    "build:replace": "node scripts/replace_protos.js",
```
Comment (Bobgy) [inline package.json:24]: nit: what do you think about calling this `update:protos`?

File: scripts/replace_protos.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  }
+  console.log("Step#1 finished: deleted two original files!");
+} catch(err) {
+  console.error(err)
+}
+
+// Step#2 download two new proto files
+const new_metadata_store_file_path = "https://raw.githubusercontent.com/google/ml-metadata/v" + mlmd_version + "/ml_metadata/proto/metadata_store.proto";
+const new_metadata_store_service_file_path = "https://raw.githubusercontent.com/google/ml-metadata/v" + mlmd_version + "/ml_metadata/proto/metadata_store_service.proto";
+download(
```
Comment (Bobgy) [inline scripts/replace_protos.js:27]: I can see you have put some efforts implementing these. They are great and does the job!
I don't think we need to change.

Just a note: If I were implementing the script, I would write the script in bash instead, tools like `wget` or `curl` can all download the file in one line. It's possible running a bash script from `package.json/scripts` too.

So far, most of the developing scripts assume developer has access to either mac or linux. It's okay for us to not accommodate for windows.

**Comment by Bobgy** [comment]
/lgtm
/approve
This is good enough!
Thank you @rui5i!

File: scripts/replace_protos.js (hunk 4 above)
Comment (rui5i) [reply scripts/replace_protos.js:27]: Thanks for the note! Learned a lot.

**Comment by Bobgy** [comment]
This was also part of https://github.com/kubeflow/pipelines/issues/4565

**Pull request merged**